src/services/projects.js

```diff
@@ -608,7 +601,70 @@ module.exports = class ProjectsHelper {
 				}
 			})
 
-			await resourceQueries.updateOne({ id: projectData.id }, { status: common.RESOURCE_STATUS_SUBMITTED })
+			//update the reviews and resource status
+			// if resource status is draft update to submitted no review changes
+			let resourceStatus = common.RESOURCE_STATUS_SUBMITTED
+			if (projectData.status === common.RESOURCE_STATUS_DRAFT) {
+				resourceStatus = common.RESOURCE_STATUS_SUBMITTED
+			} else if (
+				projectData.status === common.RESOURCE_STATUS_IN_REVIEW ||
+				projectData.status === common.RESOURCE_STATUS_SUBMITTED
+			) {
+				//update the reviews table status
+				await reviewsQueries.update(
+					{
+						resource_id: projectData.id,
+						status: common.REVIEW_STATUS_REQUESTED_FOR_CHANGES,
+						organization_id: projectData.organization_id,
+					},
+					{
+						status: common.REVIEW_STATUS_CHANGES_UPDATED,
+					}
+				)
+
+				resourceStatus = common.RESOURCE_STATUS_IN_REVIEW
+			}
+
+			//if review is not required direct publish
+			const orgConfig = await configService.list(userDetails.organization_id)
+			const orgConfigList = _.reduce(
+				orgConfig.result,
+				(acc, item) => {
+					acc[item.resource_type] = item.review_required
+					return acc
+				},
+				{}
+			)
+
+			let updateObj = {}
+
+			if (!orgConfigList[common.PROJECT]) {
+				updateObj.status = common.PUBLISHED
+				updateObj.published_on = new Date()
+
+				//update resource with published_on and status
+				await resourceQueries.updateOne({ id: projectData.id }, updateObj)
+
+				//publish the resource in consumption side
+				if (process.env.CONSUMPTION_SERVICE != common.SELF) {
```

Comment:
@priyanka-TL Create a function which will take a resource id and push it to Kafka and  API. It will be used here and in some other place as well.